compaction/compaction_manager.cc

```diff
@@ -1006,7 +1006,7 @@ class compaction_manager::offstrategy_compaction_task : public compaction_manage
             if (desc.sstables.empty()) {
                 // at this moment reshape_candidates contains a set of sstables ready for integration into main set
                 auto desc = sstables::compaction_completion_desc{ .old_sstables = std::move(old_sstables), .new_sstables = std::move(reshape_candidates)};
-                co_await t.update_sstable_lists_on_off_strategy_completion(std::move(desc));
+                co_await t.as_table_state().on_compaction_completion(std::move(desc), sstables::offstrategy::yes);
```

Comment:
This seems to also introduce a `_t.on_compaction_completion(std::move(desc));` that wasn't here before (through the new `on_compaction_completion()`). I don't know if this is a problem or not, but please clarify.

Reply:
The idea is that we'll have a single on-completion interface for both "in-strategy" and off-strategy compactions, so not to pollute table_state with one interface for each. replica::table::on_compaction_completion is being moved into private namespace.